nautobot/core/management/commands/send_installation_metrics.py

```diff
@@ -0,0 +1,67 @@
+import hashlib
+import platform
+import requests
+import uuid
+
+from constance import config
+from django.apps import apps
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from nautobot.utilities.config import get_settings_or_config
+
+
+METRICS_ENDPOINT = "https://nautobot.cloud/api/nautobot/installation-metric/"
+
+
+class Command(BaseCommand):
+    help = "Send installation metrics for this Nautobot installation."
+
+    def _hash(self, plaintext):
+        return hashlib.sha256(plaintext.encode("utf8")).hexdigest()
+
+    def get_hashed_plugins_with_version(self):
+        plugins = {}
+        for plugin_name in settings.PLUGINS:
+            plugin_name = plugin_name.rsplit(".", 1)[-1]
+            plugin_config = apps.get_app_config(plugin_name)
+            plugins[self._hash(plugin_name)] = getattr(plugin_config, "version", None)
+        plugins = dict(sorted(plugins.items()))
+
+        return plugins
+
+    def handle(self, *args, **options):
+        # skip if metrics are disabled
+        if settings.INSTALLATION_METRICS_ENABLED is not True:
+            self.stdout.write("Installation metrics are disabled by INSTALLATION_METRICS_ENABLED setting, skipping.")
+
+        # get the deployment id for this install from constance or settings
+        # if one is not already set, generate a random uuid and set it in constance
+        deployment_id = get_settings_or_config("DEPLOYMENT_ID")
```

Comment:
Do we actually need/want to support setting `DEPLOYMENT_ID` in settings? Or should it always come from constance only?

Reply:
I can see the utility in setting this as a config item in deployment. How else would you give a deployment ID to a fresh install? If Constance-only, an updated deployment ID wouldn't be sent until `post_upgrade` is re-run. I think this would incur a lot of false reports.